manim/mobject/number_line.py

```diff
@@ -202,34 +392,82 @@ def get_number_mobject(
         num_mob.next_to(self.number_to_point(number), direction=direction, buff=buff)
         return num_mob
 
-    def get_number_mobjects(self, *numbers, **kwargs):
+    def get_number_mobjects(self, *numbers: float, **kwargs) -> VGroup:
+        """Returns the labels (type :class:`~.DecimalNumber`) of the number line as a VGroup.
+
+        Parameters
+        ----------
+        numbers : :class:`list[float]`
+            the numbers to put
+
+        Returns
+        --------
+        :class:`~.VGroup`
+            VGroup containing the :class:`~.DecimalNumber` mobjects.
+        """
         if len(numbers) == 0:
             numbers = self.default_numbers_to_display()
         return VGroup(
             *[self.get_number_mobject(number, **kwargs) for number in numbers]
         )
 
-    def get_labels(self):
+    def get_labels(self) -> VGroup:
+        """Returns the default labels (type :class:`~.DecimalNumber`) of the number line as a VGroup.
+
+        Returns
+        --------
+        :class:`~.VGroup`
+            VGroup containing the :class:`~.DecimalNumber` mobjects.
+        """
         return self.get_number_mobjects()
 
-    def add_numbers(self, *numbers, **kwargs):
+    def add_numbers(self, *numbers: float, **kwargs) -> "NumberLine":
+        """Add the numbers (labels) to the NumberLine
+
+        Parameters
+        ----------
+        numbers : :class:`list[float]`
+            the numbers to put
+
+        Returns
+        -------
+        :class:`NumberLine`
+            The current number line object (self).
+        """
         self.numbers = self.get_number_mobjects(*numbers, **kwargs)
         self.add(self.numbers)
         return self
 
 
 class UnitInterval(NumberLine):
+    """Creates a NumberLine with a unit interval """
+
     def __init__(
         self,
-        unit_size=6,
-        tick_frequency=0.1,
-        numbers_with_elongated_ticks=[0, 1],
-        number_at_center=0.5,
-        decimal_number_config={
+        unit_size: int = 6,
+        tick_frequency: float = 0.1,
+        numbers_with_elongated_ticks: typing.List[float] = [0, 1],
+        number_at_center: float = 0.5,
+        decimal_number_config: dict = {
```

Comment:
```suggestion
        decimal_number_config: typing.Dict[sstr,int] = {
```

Reply:
I think u have a typo here and I am not sure it is correct typing annotation in the first place. See above comment.